packages/lix-sdk/src/branch/merge-branch.ts

```diff
@@ -0,0 +1,171 @@
+import type { Branch, Change } from "../database/schema.js";
+import type { Lix } from "../open/openLix.js";
+import type { DetectedConflict } from "../plugin/lix-plugin.js";
+import { getLowestCommonAncestorV2 } from "../query-utilities/get-lowest-common-ancestor-v2.js";
+
+export async function mergeBranch(args: {
+	lix: Lix;
+	sourceBranch: Branch;
+	targetBranch: Branch;
+}) {
+	const executeInTransaction = async (trx: Lix["db"]) => {
+		const diffingPointers = await getBranchChangePointerDiff(trx, args);
+
+		// could be queried in a single query with a join
+		// but decided to keep it simpler for now
+		const diffingChanges = await trx
+			.selectFrom("change")
+			.select(["id"])
+			.where(
+				"id",
+				"in",
+				diffingPointers.map((pointer) => pointer.source_change_id),
+			)
+			.selectAll()
+			.execute();
+
+		const plugins = await args.lix.plugin.getAll();
+		const detectedConflicts: DetectedConflict[] = [];
+
+		// let plugin detect conflicts
+		await Promise.all(
+			plugins.map(async (plugin) => {
+				if (plugin.detectConflictsV2) {
+					const conflicts = await plugin.detectConflictsV2({
+						lix: args.lix,
+						changes: diffingChanges,
+					});
+					detectedConflicts.push(...conflicts);
+				}
+			}),
+		);
+
+		// update the branch change pointers for non-conflicting changes
+		await Promise.all(
+			diffingPointers.map(async (pointer) => {
+				// potentially expensive query
+				const pluginDetectedConflict = detectedConflicts.find(
+					(conflict) =>
+						conflict.change_id === pointer.target_change_id ||
```

Comment:
how can we have conflicts where the change_id is pointing to the target_change_id?

Reply:
fixed